content/articles/working-with-github-web-api-in-python/index.md

```diff
@@ -0,0 +1,241 @@
+
+```bash
+Status code: 200
+dict_keys(['items', 'total_count', 'incomplete_results'])
+```
+
+The request was completed since the `status_code` is `200`.
+
+### Using the response dictionary
+We can interact with the data contained in the dictionary since the result from the API call has been saved. Let's make a report that sums up the information. This is a cool method to ensure we got the information we wanted and to get started looking at the information we care about:
+
+```python
+import requests
+
+# Create an API call 
+url = 'https://api.github.com/search/repositories?q=language:python&sort=stars'
+j = requests.get(url)
+print("Status code:", j.status_code)
+# In a variable, save the API response.
+respons_dict = j.json()
+print("Total repos:", respons_dict['total_count'])
+# Find out more about the repositories.
+repos_dicts = respons_dict['items']
+print("Repos found:", len(repos_dicts))
+# Examine the first repository.
+repos_dict = repos_dicts[0]
+print("Keys:", len(repos_dict))
+for key in sorted(repos_dict.keys()):
+ print(key)
+```
+The value linked with `total_count`, which reflects the amount of GitHub Python projects, is first printed.
+
+The value of `items` is a list of dictionaries, each of which provides information about a single Python repository.
+
+The list of dictionaries is then saved in `repos_dicts`. The length of `repos_dicts` is then printed to check  how many of repositories we have their information.
+
+We select the first item from `repos_dicts` and put it in `repos_dict` to look more closely at the information given about each repository. To see what information we have, we print the amount of keys in the dictionary. Finally, we write each of the dictionary's keys to check what type of data is contained.
+
+The output begins to paint a more detailed picture of the data.
+
+```bash
+Status code: 200
+Total repos: 7694326
+Repos found: 30
+Keys: 74
+archive_url
+archived
+assignees_url
+--snip--
+
+url
+watchers
+watchers_count
+```
+
+The GitHub API gets back a range of data for every repository: `repos_dict` has `74` keys. You may get a sense of the type of information you can get about a project by observing these keys.
+
+Let's have a look at what some of the keys in repos dict entail:
+
+```python
+import requests
+
+# Create an API call 
+url = 'https://api.github.com/search/repositories?q=language:python&sort=stars'
+j = requests.get(url)
+print("Status-code:", j.status_code)
+# In a variable, save the API response.
+respons_dict = j.json()
```

Comment:
Remove these too.

Add new set of codes in a code snippet.

You can write the consolidated version at the end or put them in a GitHub repo, and paste the link in the conclusion